lib/ui/window/viewport_metrics.h

```diff
@@ -9,7 +9,11 @@
 
 namespace flutter {
 
-static const double kUnsetDepth = -1;
+// This is the value of double.maxFinite from dart:core.
+// Platforms that do not explicitly set a depth will use this value, which
+// avoids the need to special case logic that wants to check the max depth on
+// the Dart side.
+static const double kUnsetDepth = 1.7976931348623157e+308;
```

Comment:
(Nothing to resolve) I'm a little concerned about if this could ever be potentially larger than `std::numeric_limits<double>::max()`, and what would happen if it was. I'm not sure if that's an actual practical problem that we would expect to see sometimes or not though. From some cursory investigation it looks like it should be fine.

Reply:
Would that not result in a compilation error?

This number is actually smaller than what a 64 bit double should be able to hold from what I can tell.

Reply:
That would make sense to me, but I think it's undefined behavior and compiler dependent. In some cases I'm pretty sure it'll get silently truncated to whatever will fit. From what I could track down this should be fine going by the IEEE 754 standard for 64 bit types, which is "usually" how double is implemented in C++. So it's most likely good.